08-rails/database-intro.md

```diff
@@ -15,32 +15,58 @@ As defined by wikipedia:
+A relational database consists of structured data, organized in tables. Data in each table is stored in _rows_ and _columns_. 
 
-#### Vocabulary
-- **Schema**: A representation of the structure of a database. It may be graphical or textual.
-- **Table**: A collection of closely related columns. A table consists of rows each of which shares the same columns but vary in the column values.
-- **Column**: A single unit of named data that has a particular data type. Columns only exist in tables.
-- **Row**: One set of related values for all of the columns declared in a given table. Also known as a record.
-- **Primary Key**: A column or group of columns in a given table that uniquely identify each row of the table.
-- **Foreign Key**: One or more columns in a table intended to contain only values that match the related primary/unique key column(s) in the referenced table.
+In addition, these tables can have relationships with other tables which are _formally described_, meaning the database system will enforce the "rules" defined by the relationships we setup.
+
+### So...what about the CSV files we used?
+The CSV files we used in the `Ride Share Rails` project effectively acted as our database. They persisted data about drivers, passengers, and trips. This is why when we run our program and add a trip, the trip will remain in our system, even after restarting the program.
+
+CSV files do act a whole lot like Relational Databases. Each CSV file can act as it's own table. Each file has multiple columns and rows. Just like in a Relational Database, the rows each represent a unique item in the table and the columns represent different attributes each item has.
+
+This is what our Drivers CSV file looked like from our Ride Share Rails project. The `name` column will tell us what each driver's name is and the `row` with `id` of 1 will tell us the information we need for one specific driver. (In this case, Paul Klee.)
+![Drivers CSV](./images/drivers_csv.png)
+
+#### So why bother working with Relational Databases when CSV files do the job just fine?
+Glad you asked! There are actually a number of limitations CSV files have that RDBs do not:
+- We can't enforce relationships. 
+  - As mentioned above RDBs will actually enforce rules we create about relationships between our tables so we don't end up with mis-matched data. But this is a whole lesson on it's own. We'll get to it soon enough
+- We can't scale enough. 
+  - CSVs have worked well for us when our system is only tracking 100 trips between a few different drivers and passengers but it is not designed to handle millions or billions of rows in hundreds of different tables. Most enterprise software is operating at that scale or is planning to operate at that scale in the future. RDBs were designed to handle this scale.
```

Comment:
You might also include a note about how RDBs can manage concurrent access from different servers.  A CSV can only be edited by one process at a time.

Reply:
Hmmm, my understanding is that for both csv files and RDBs concurrency handling would be dependent on the system handling it (Google Drive, Cassandra, etc.). 
I know little enough about this that I'm hesitant to try to add anything about that.